ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/slot/DiskSlots.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2020, Red Hat, Inc. and/or its affiliates,
+ * and individual contributors as indicated by the @author tags.
+ * See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ * This copyrighted material is made available to anyone wishing to use,
+ * modify, copy, or redistribute it subject to the terms and conditions
+ * of the GNU Lesser General Public License, v. 2.1.
+ * This program is distributed in the hope that it will be useful, but WITHOUT A
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * (C) 2020,
+ * @author Jonathan Halliday (jonathan.halliday@redhat.com)
+ */
+package com.arjuna.ats.internal.arjuna.objectstore.slot;
+
+import com.arjuna.ats.arjuna.logging.tsLogger;
+
+import java.io.*;
```

Comment:
Hmm. Should the build scripts be failing the build then? Perhaps using the "/build.sh -am -pl :narayana-jta" rather than doing a full "/build.sh" is bypassing something it shouldn't be?

Reply:
Well we caught between the proverbial rock and a hard place. We cannot checkstyle ArjunaCore since it contains very old source files and we want to be able to perform meaningful diffs with the old source (fixing checkstyle errors would mean we could no longer easily diff revisions unless the diff tool ignores whitespace etc). So instead we just ask engineers to follow our rules. But since that may present a barrier for occasional contributors, if you prefer not incur the overhead of having to manually follow these rules then we will not block your PR.

Reply:
ok, at least the build scripts aren't broken, they just aren't intended to be warning or enforcing. Is there any tool support, or do I just have to remember to eyeball the code?

Reply:
You have two options:

1. Enable the checkstyle plugin via the [pom file](https://github.com/jbosstm/narayana/blob/master/pom.xml#L377), or

2. Reference the rule file from your IDE settings (for Intelij that would be Settings -> Other Settings -> Checkstyle) and we use the [wildfly rule set](https://github.com/wildfly/wildfly-checkstyle-config/tree/master/src/main/resources/wildfly-checkstyle).

Reply:
Any advise on how we can improve upon that would be useful but I do not see any alternaive.

Reply:
I did say "if you prefer not incur the overhead of having to manually follow these rules then we will not block your PR" so if you do not plan to run checkstyle then you need to resolve this conversation.

Reply:
Sorry, got sidetracked doing a memory mapped version of the backend. Should all be sorted now.